task01/src/Vector.cpp

```diff
@@ -0,0 +1,318 @@
+#pragma once
+#include "Vector.h"
+
+
+//--------------------------------------------------------------------------------
+
+/*MOVE constructor*/
+Vector::Vector(Vector&& other) noexcept : m_size(other.m_size), m_capacity(other.m_size), m_array(other.m_array) {
+    /*std::cout << "moved" << std::endl;*/
+
+    other.m_size = 0;
+    other.m_capacity = 0;
+    other.m_array = nullptr;
+
+}
+//--------------------------------------------------------------------------------
+/*Vector class constructor*/
+Vector::Vector(std::size_t size, int value)
```

Comment:
C++ constructors use initialization lists, it is considered bad practice to set the instance's members in the c'tor's body